### PR TITLE
chore(storybook): update to Storybook 5

### PR DESCRIPTION
**Pull request by francoischalifour**
This upgrades to Storybook 5 ([see migration notes](https://github.com/storybooks/storybook/blob/next/MIGRATION.md#from-version-41x-to-50x)).

Since the [URL structure changed](https://github.com/storybooks/storybook/blob/next/MIGRATION.md#new-url-structure), we need to update the documentation story links when merging this PR.

[See stories →](https://deploy-preview-3585--instantsearchjs.netlify.com/stories)

### CODE REVIEW

**Comment by Haroenv** [comment]
The search doesn't seem be working for me 🤔

**Comment by Haroenv** [comment]
You can add redirections of a query string with Netlify, but as said, I'm not sure if it's worth it, we can just update the URLs in the docs. 

This is definitely a nice improvement!

**Comment by Haroenv** [comment]
Can the logo be changed to "InstantSearch.js" again with v5?

**Comment by francoischalifour** [comment]
> The search doesn't seem be working for me 🤔

Indeed, it works in watch mode though.
 
> You can add redirections of a query string with Netlify, but as said, I'm not sure if it's worth it, we can just update the URLs in the docs.

The previous links will keep working for a while because Storybook keeps backward compatibility. Still, we should update them in the doc.

> Can the logo be changed to "InstantSearch.js" again with v5?

Yes – which logo do you have in mind?

**Comment by Haroenv** [comment]
> Can the logo be changed to "InstantSearch.js" again with v5?

Either the text, or we can ask for / make  a horizontal version of the InstantSearch logo with the text next to it

**Comment by francoischalifour** [comment]
The search in production bug will be solved by upgrading to Storybook 5.0.1 ([see issue](https://github.com/storybooks/storybook/pull/5909)). I'll update the PR.

Regarding the logo, it could be cool to ask the design team. Right now we don't have a logo with text AFAIK.

File: package.json [hunk 1]
```diff
@@ -26,8 +26,8 @@
     "build:cjs": "BABEL_ENV=cjs babel src --extensions '.js,.ts,.tsx' --out-dir cjs/ --ignore 'src/index.es.js','**/__tests__','**/__mocks__' --quiet",
     "build:es": "BABEL_ENV=es babel src --extensions '.js,.ts,.tsx' --out-dir es/ --ignore 'src/index.es.js','**/__tests__','**/__mocks__' --quiet && BABEL_ENV=es babel src/index.es.js --out-file es/index.js --quiet",
     "doctoc": "doctoc --no-title --maxlevel 3 README.md CONTRIBUTING.md",
-    "storybook": "start-storybook -p 6006 --ci --static-dir .storybook/static",
-    "storybook:build": "build-storybook --output-dir website/stories --static-dir .storybook/static",
+    "storybook": "start-storybook --quiet --port 6006 --ci --static-dir .storybook/static",
```
Comment (francoischalifour) [inline package.json]: I added quiet modes to reduce the unnecessary pollution in the terminal. This option in the `storybook:build` script should also speed up the build time on Netlify.

**Pull request merged**